rust/actyx/ax-core/src/private_key.rs

```diff
@@ -10,12 +10,33 @@ use crate::{
     util::formats::{ActyxOSCode, ActyxOSError, ActyxOSResult, ActyxOSResultExt},
 };
 use libp2p::identity;
+use rand::RngCore;
 
-use crate::{certs::DeveloperCertificate, cmd::get_data_dir};
+use crate::certs::DeveloperCertificate;
 
 const PUB_KEY_FILE_EXTENSION: &str = "pub";
 pub const DEFAULT_PRIVATE_KEY_FILE_NAME: &str = "id";
 
+// NOTE: I'm not sure where to put this
+/// Generate a Swarm Key
+pub fn generate_key() -> String {
+    let mut key = [0u8; 32];
+    rand::thread_rng().fill_bytes(&mut key);
+    base64::encode(key)
+}
+
+/// Returns the data directory for Actyx. Does not create the folders!
+/// https://docs.rs/dirs/3.0.1/dirs/fn.config_dir.html
+///
+/// Platform    Value                               Example
+/// Linux       $XDG_CONFIG_HOME or $HOME/.config   /home/alice/.config
+/// macOS       $HOME/Library/Application Support   /Users/Alice/Library/Application Support
+/// Windows     {FOLDERID_RoamingAppData}           C:\Users\Alice\AppData\Roaming
+fn get_data_dir() -> ActyxOSResult<PathBuf> {
```

Comment:
Probably better in `util`, but overall,
I think `ax-core` shouldn't care about paths at all, and this better be put in `ax`, consequently default loading mechanism shouldn't be written inside `ax-core`, instead it should be supplied by `ax` e.g. via dependency-injection

Reply:
I forgot to reply. I don't think addressing this right now is pertinent, though I agree. Both this and the previous issue you outlined are relevant but a bit outside the scope of this PR. Should we open an issue?

Reply:
https://github.com/Actyx/Actyx/issues/618
^ Absolutely. There's no need to fix this in this PR I think